config/kernel-bio-op.m4

```diff
@@ -65,3 +85,28 @@ AC_DEFUN([ZFS_AC_KERNEL_BIO_BI_OPF], [
 		AC_MSG_RESULT(no)
 	])
 ])
+
+dnl #
+dnl # Linux 4.10 API,
+dnl #
+dnl # bio_set_op_attrs() was changed to an inline function instead
+dnl # of a macro.
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_BIO_SET_OP_ATTRS_IS_FUNCTION], [
+	AC_MSG_CHECKING([whether bio_set_op_attrs is a function])
+	ZFS_LINUX_TRY_COMPILE([
+		#include <linux/blk_types.h>
+	],[
+		#ifndef bio_set_op_attrs
+		(void) bio_set_op_attrs(NULL, 0, 0);
+		#else
+		#error "bio_set_op_attrs is a macro"
```

Comment:
My rationale here is that there are apparently 3 cases possible with respect to ```bio_set_op_attrs```:  1. it's defined as a macro, 2. it's defined as an inline function, 3. it's not defined at all.  In case 1, we want to use the kernel-defined macro.  In cased 2, we want to use the inline function.  In case 3, we want to define it ourselves in terms of manipulation of ```bi_rw``` (which hopefully actually exists).  I'm not quite sure how else to handle this and get the right definition in ```vdev_disk.c``` where it's used other than this odd combination of an autotools check and the conditional compilation.